crates/node/core/src/node_config.rs

```diff
@@ -422,6 +422,29 @@ impl<ChainSpec> NodeConfig<ChainSpec> {
             Err(e) => Err(eyre!("Failed to load configuration: {e}")),
         }
     }
+
+    /// Type erases the [`ChainSpec`] generic into a [`Box<dyn EthChainSpec>`].
+    pub fn erased(self) -> NodeConfig<Box<dyn EthChainSpec>>
```

Comment:
what we want here is `fn map_chainspec<F, C> where F:FnOnce(Self) -> NodeConfig<C>`

Reply:
ahhh got it sorry I misunderstood what you wanted, thought you just wanted to extract the erasing part. Updating.